Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 17, in <module>
    my_dog=Dog('十六岁',8)
TypeError: Dog() takes no arguments
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 17, in <module>
    my_dog=Dog('十六岁',8)
TypeError: Dog() takes no arguments
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 17, in <module>
    my_dog=Dog(十六,8)
NameError: name '十六' is not defined
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 17, in <module>
    my_dog=Dog("十六",8)
TypeError: Dog() takes no arguments
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 17, in <module>
    my_dog=Dog("十六",8)
TypeError: Dog() takes no arguments
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 17, in <module>
    my_dog=Dog("hj",8)
TypeError: Dog() takes no arguments
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 17, in <module>
    my_dog=Dog("",8)
TypeError: Dog() takes no arguments
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========

>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========

>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
ijj
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
<bound method Dog.sit of <__main__.Dog object at 0x0000029746149908>>
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
ijj
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 19, in <module>
    print(o)
NameError: name 'o' is not defined
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
Wangis now sitting
None
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
Wangis now sitting
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
Wangis now sitting
None
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
6942
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
<bound method Dog.sit of <__main__.Dog object at 0x000002099DFA8DC8>>
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
Wangis now sitting
None
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 19, in <module>
    k=my_dog.sit(11)
TypeError: sit() takes 1 positional argument but 2 were given
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 19, in <module>
    k=my_dog.sit(11)
TypeError: sit() takes 1 positional argument but 2 were given
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
Wangis now sitting
None
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
8
Wangis now sitting
None
wangkaixuan
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
wangkaixuan
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 21, in <module>
    print(my_new_car.get_descriptive_name())
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 15, in get_descriptive_name
    long_name=str(sekf.yaer)+' '+self.make+self.mosel
NameError: name 'sekf' is not defined
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
wangkaixuan
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 21, in <module>
    print(my_new_car.get_descriptive_name())
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 15, in get_descriptive_name
    long_name=str(sekf.yaer)+' '+self.make+self.modle
NameError: name 'sekf' is not defined
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
wangkaixuan
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 21, in <module>
    print(my_new_car.get_descriptive_name())
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 15, in get_descriptive_name
    long_name=str(self.yaer)+' '+self.make+self.modle
AttributeError: 'Car' object has no attribute 'modle'
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
wangkaixuan
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 21, in <module>
    print(my_new_car.get_descriptive_name())
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 15, in get_descriptive_name
    long_name=(str(self.yaer)+' '+self.make+self.modle)
AttributeError: 'Car' object has no attribute 'modle'
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
wangkaixuan
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 21, in <module>
    print(my_new_car.get_descriptive_name())
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 15, in get_descriptive_name
    long_name=(str(self.yaer)+' '+self.make+self.modle)
AttributeError: 'Car' object has no attribute 'modle'
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
wangkaixuan
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 21, in <module>
    print(my_new_car.get_descriptive_name())
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 15, in get_descriptive_name
    long_name=(str(self.yaer)+' '+self.make+self.modle)
AttributeError: 'Car' object has no attribute 'modle'
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
wangkaixuan
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 21, in <module>
    print(my_new_car.get_descriptive_name())
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 15, in get_descriptive_name
    long_name=(str(self.yaer)+' '+self.make+self.modle)
AttributeError: 'Car' object has no attribute 'modle'
>>> 
>>> 
>>> 
>>> 
>>> 
>>> 
>>> 
>>> 
>>> 
>>> 
>>> 
>>> 
>>> 
>>> 
>>> 
=========== RESTART: C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py ===========
wangkaixuan
Traceback (most recent call last):
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 21, in <module>
    print(my_new_car.get_descriptive_name())
  File "C:/Users/86666/Desktop/自己编写的Python程序/类的知识.py", line 15, in get_descriptive_name
    long_name=(str(self.yaer)+' '+self.make+self.modle)
AttributeError: 'Car' object has no attribute 'modle'
>>> help
